### ISSUE TITLE
Sub can't move laterally

### ISSUE THREAD

**Issue by lphillips** [Regalis11/Barotrauma#3624]
In a multiplayer session, I ran into an issue where the sub was no longer able to move forward or backward. It occurred after removing power to the engine by rewiring the junction box. After restoring power to the engine, the sub would no longer move forward or backward, but the engine did indicate that it had power and thrust, and the bubble animation was playing on the engine indicating it should be moving. 

Reproduction steps:
1) Remove power to the engine by removing the power wire from the junction box.
2) Wait 5 - 10 minutes (might not be necessary, but it was what was done in the sessions where this bug was reproduced).
3) Reattach power wire for engine to junction box.
4) Attempt to move sub

The sub should be able to move, but doesn't. I was able to reproduce this bug twice following these steps. I also observed a somewhat similar bug by removing the power wire for the ballast pumps and reattaching it later. The pumps began operating, but were out of sync and the water level would keep resetting to 100% in the ballast tank. 

Other information that may be useful:

- Bug occurred in multiplayer session, in campaign mode, with 4 players and no bots
- Using a Dugong with 1 level in Hull upgrade, 1 level in Engine Efficiency upgrade, and 1 level in Gun View Range upgrade
- Ship had no known damage when bug occurred

Running on version 0.10.4.0 on Windows

**Comment by Udrakan** [comment]
Can confirm the ballast problem. Water appears to drain fully, then reset to 100% in an instant.

**Comment by Regalis11** [comment]
I'm afraid I wasn't able to reproduce this. Not sure if there's much more info you can provide, but if anyone figures out any additional steps or details that might help in getting this to occur, please let us know.

**Comment by Regalis11** [comment]
I found one issue that could potentially be related to this. The server only syncs the state of the reactor when someone adjusts the settings, and if the state gets desynced for whatever reason, the clients will not be notified until someone operates the reactor. So it could be possible that in this case the wiring was not the problem, but the sub was unable to move because it was powered down server-side while appearing to be powered client-side.

Now the server updates the state of the reactor regularly even if no-one operates it: https://github.com/Regalis11/Barotrauma-development/commit/95c9a6a

**Comment by alkaid7205** [comment]
I was able to to reproduce this (engine issue) by following the same steps I showed in the video in the #3952 by unplugging the engine instead of the ballast. 
1. Host a server and start any mission (undock from the outpost to test the engine)
2. Disconnect the power wire entirely both sides (have it in your inventory) from the engine and its JB.
3. Wire the cable from the engine to a different junction box.
4. Try moving the sub.
**result:**
Engine is on 100% thrust (in the command panel and in the engine panel) but the sub isn't moving.
